spec/puck/jar_spec.rb

```diff
@@ -147,9 +147,21 @@ def resolve_gem_dependencies(options)
             @tmp_dir = Dir.mktmpdir
           end
 
-          it 'includes extra files' do
-            create_jar(@tmp_dir, dependency_resolver: dependency_resolver, extra_files: %w[config/app.yml])
-            jar_entries.should include('META-INF/app.home/config/app.yml')
+          context 'with extra files' do
+            it 'preserves relative paths for array argument' do
+              create_jar(@tmp_dir, dependency_resolver: dependency_resolver, extra_files: %w[config/app.yml])
+              jar_entries.should include('META-INF/app.home/config/app.yml')
+            end
+
+            it 'is possible to include files using globs' do
+              create_jar(@tmp_dir, dependency_resolver: dependency_resolver, extra_files: %w[config/*.yml])
+              jar_entries.should include('META-INF/app.home/config/app.yml', 'META-INF/app.home/config/another.yml')
+            end
+
+            it 'uses specified paths for hash argument' do
```

Comment:
I don't understand this description. I think you mean this:

```ruby
context 'when the argument is a hash'
  it 'uses the keys for the path of the content and the value as the path within the JAR'
```

Maybe a test that `nil` works as value too? I know it's covered by the previous tests in practice, but that's just because the code looks the way it does (that you cleverly use a quirk of Ruby and `#each` to make it work the same for both types). If the code was implemented differently, say by having different paths for array and hash values, it would no longer be covered.